Source/Libraries/openXDA.APIAuthentication/APIQuery.cs

```diff
@@ -261,6 +264,12 @@ private HttpRequestMessage BuildRequest(Host host, string path, Action<HttpReque
 
                 const string type = "XDA-API";
                 string decode = $"{APIKey}:{APIToken}";
+
+                if (!(user is null))
+                {
+                    decode = $"{decode}:{user.Identity.Name}";
```

Comment:
I guess the intent here is to pass in the `IPrincipal` of the user that is currently authenticated in the local system and send their username to the remote system to be looked up based on `AdoSecurityProvider.RefreshData()`? I guess it might be pretty strange to assume that a given user might have different usernames across systems or that two different users might have the same username across systems.

That said, if two users have the same name in real life, this could represent a potential security hole. I don't feel the risk is necessarily high enough that we need to change it. Just something to be aware of.

Reply:
the problem is we don't really have a way of checking. For Database Users we could pass the `ID` but for AD and Azure we rely on the fact that both systems are tied into the same Azure and Active Directory Systems.